From cc14b371c38cec4c35530f95078713e1e368c6c7 Mon Sep 17 00:00:00 2001
From: Andrea Adami <andrea.adami@gmail.com>
Date: Mon, 14 Jun 2021 23:49:35 +0200
Subject: [PATCH 1/1] libtuxtxt: fix build with musl

fix
libtuxtxt.c:47:46: error:
 'PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP'
 undeclared here (not in a function);
  did you mean 'PTHREAD_MUTEX_INITIALIZER'?

Signed-off-by: Andrea Adami <andrea.adami@gmail.com>
---
 libtuxtxt.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libtuxtxt.c b/libtuxtxt.c
index 476c7db..d7870dc 100644
--- a/libtuxtxt.c
+++ b/libtuxtxt.c
@@ -44,7 +44,12 @@
  ******************************************************************************/
 
 static int tuxtxt_initialized=0;
+
+#ifdef __GLIBC__
 static pthread_mutex_t tuxtxt_control_lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
+#else
+static pthread_mutex_t tuxtxt_control_lock = {{PTHREAD_MUTEX_RECURSIVE}};
+#endif
 static pthread_mutex_t tuxtxt_key_queue_lock = PTHREAD_MUTEX_INITIALIZER;
 
 int tuxtxt_init()
-- 
2.17.1

